sys/linux/dev_kvm.txt

```diff
@@ -261,10 +268,20 @@ syzos_api_msr {
 	arg_value	int64
 }
 
+kvm_smc_id = 0x80000000, 0x40000000, 0x1000000, 0x2000000, 0x3000000, 0x4000000, 0x30000000, 0x31000000, 0x32000000, 0x3f000000, 0xffff, 0x0, 0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7, 0x8, 0x10, 0x20, 0x40, 0x80, 0x100, 0x200, 0x400, 0x800, 0x1000, 0x2000, 0x4000, 0x8000, 0x10000
+
+syzos_api_smc {
+	call		const[3, int64]
+	size		bytesize[parent, int64]
```

Comment:
At some point it would be good to remove this duplication with a template:
```
type syzos_api[NUM, PAYLOAD] {
	call		const[NUM, int64]
	size		bytesize[parent, int64]
	payload		PAYLOAD
}

syzos_api_smc {
	arg_id		flags[kvm_smc_id, int32]
	arg_params	array[int64, 5]
}
```

Then some calls may not even need a separate struct, if it's just int/flags, e.g. uexit:
```
syzos_api_call [
	uexit	syzos_api[0, intptr]
	code	syzos_api[1, syzos_api_code]
	msr	syzos_api[2, syzos_api_msr]
	smc	syzos_api[3, syzos_api_smc]
] [varlen]
```

Reply:
Good idea! I can probably fix that right away.

Reply:
Added another patch to the series.